### ISSUE TITLE
V7 Reboot Draft Feedback

### ISSUE THREAD

**Issue by Riven202** [LogicAndTrick/snarkpit-reboot#2]
## General Comments:
- Loving the responsiveness of the new site! Definitely makes it feel much more modern!
### Font Size:
- Font size seems a little small. It's at 0.8rem from what I can tell, can that be a smidgen larger? 
- Looks like TWHL body font size is around 14pt on my 4K 34" screen which feels about right. 
   -  ![image](https://github.com/LogicAndTrick/snarkpit-reboot/assets/1342806/c205d606-574f-4dc9-8c61-f451a729408d)
- Do we need to add the GDPR cookie notice onto the Pit? 
### Mobile Sliding Sheet Menu: 
- I'm lovin' the new mobile sheet menu! That's awesome! Since functionally it does exactly what it needs to, can we dress it up to appear less stock? 
   - Give each link the same yellow link style as other links on the site. 
   - Can the text links be slightly larger to feel like top-level menu links? Perhaps larger than a revised 14pt body font size?
   - Love the inline search bar. That's slick! 
   - Can we replace the Snarkpit text at the top with the image logo? (I may work on a vector .svg version of the logo if possible to replace the text portion potentially). 

## Home Page:
- Content not centered on mobile screen size: 
   - ![image](https://github.com/LogicAndTrick/snarkpit-reboot/assets/1342806/fdf4fb49-cfaf-4c5e-bf80-cb94f7187a6e)
-  Mobile screen size changes:
    - Place Community News section first at mobile screen size, followed-by:
       - Recent Forum Posts
       - Spotlight (see notes below)
       - Latest Maps
       - Recent Updates
       - Latest Journals
- At the medium screen size, move Latest Journals section beneath Latest Maps Section
   - ![image](https://github.com/LogicAndTrick/snarkpit-reboot/assets/1342806/c18a7322-710d-4666-b9c8-86c3e9f0ca76)
- Spotlight Maps Changes: 
   - I really like how the latest maps are shown in a combined view. Can the spotlight view also be shown this way at the mobile screen resolution breakpoints?  
   - ![image](https://github.com/LogicAndTrick/snarkpit-reboot/assets/1342806/6febbfd4-fbf8-4eef-a0e2-ab8e3b8837f7)

## Maps Landing Page:
- Love the instant filtering. It's MUCH speedier now, and appreciate the new "upload a map" button location. This will make it much easier to find. 
- Are star maps and star map sorting still available? As seen from the old site: 
   - ![image](https://github.com/LogicAndTrick/snarkpit-reboot/assets/1342806/55dbda13-67da-4c23-92e6-6e237ccb6981)
- Map Item Containers:
   - This is not anything on your layout, and was always a design element in the original Pit that bugged me, where we had these large gaps at non-standard image sizes for maps and other image elements that left large gaps as shown below: 
   - ![image](https://github.com/LogicAndTrick/snarkpit-reboot/assets/1342806/476b35b1-8b3d-4672-8d80-1abe6f554bc4)
- Can these be containerized in the way the Spotlight map items are contained within a larger frame? 
   - That, or achieve something similar with filler color and middle-aligned images such as on TWHL's vault: 
      - Spotlight container edges look cleaner:
      - ![image](https://github.com/LogicAndTrick/snarkpit-reboot/assets/1342806/78f236de-3b42-4e87-8166-7e754602b55e)
   - TWHL's filler bars are a nice fix:
      - ![image](https://github.com/LogicAndTrick/snarkpit-reboot/assets/1342806/8d1803f9-5731-4d9a-b94a-2068460a45b3)
- Does editing a map's detail page sort it to the top of the newest first list? (not ideal). However,  creating an option to sort by "recently updated" may be useful in that way. 

## Map Detail Page:
- The inline enlarged image view was desperately needed. Thank you for adding that!
- Map description text is nice to see without needing to scroll now. 
- Is it difficult to keep an old "watch discussion" button/link for a map? Shown from the old site below:
   - ![image](https://github.com/LogicAndTrick/snarkpit-reboot/assets/1342806/d64c9400-6ee2-4cef-a446-04bdfb3b2224)
- Can we rename the download button from "Download" to "SnarkPit Download" ?
   - This is my effort to highlight that we host our own map files, no need to upload elsewhere. 
- Optional New Admin feature request: 
   - Can we add a link to the game store / download page? 
      - I want to make it easier for folks to find the actual game, so they can play the mod/map easily. 
      - Can we rename the circled link below from "game:" to "More Content for:" 
      - And then add a new line beneath that titled: "Get game:" and a field link of the game title, populated from an admin table I can edit to include that external game link / store page for each game database entry?
      - ![image](https://github.com/LogicAndTrick/snarkpit-reboot/assets/1342806/0fb94125-e5fd-4c97-a60b-19ac07ddc341)

## Articles / Downloads Landing Page: 
- At the largest resolution, can we reduce the size of each row thumbnail image item to a max size of 218 px wide (or around that size?) I think 308px is probably too large for the size of those original images. 

## User Detail Page: 
- See notes under Maps Landing Page for Map container Item comments (above). 

### Group Landing Page: 
- Is it easy to add back in the "Other Users online" link and short list?
### Forums Thread Landing Page:
- Are the snark icons used to represent an individual thread still active (the hot thread icon, etc.)? 
   - Is there a legend or way to see what those represent? Or instead of a legend, just adding an alt text on hover might be enough. 
### Forum Thread Page:
- I really like the centered usernames and blurbs for Individual posts, but can't help but notice the different padding. Please revise slightly: 
   - ![image](https://github.com/LogicAndTrick/snarkpit-reboot/assets/1342806/5ccff6d1-befb-4e6c-96c3-4578c1b37c65)
- I REALLY appreciate the new anchor links to individual threads. That would have been soooo useful back in the day. 
- Love the new Reply panel! 
- It appears BBcode was finally replaced. Does that mean the site is already parsing that out?
   - formatting help page does not yet appear for me. 
   - I like the new formatting buttons and syntax! Does it make sense to use a Markdown parser instead, and replace those functions with markdown, since that seems to be the way most of the internet thread formatting is going? 
   - Smilies appear broken in the live preview pane. 

More comments to come!

**Comment by LogicAndTrick** [comment]
Glad you're liking it overall, and this is great feedback. Going to make a checklist for myself to work on over the week. Also a few responses to questions/etc:

- Increasing the font size is easily done, but might break some other stuff around the site since I styled things using that font size. Not a big deal, but might need some extra fixes if I miss some.
- GDPR notice isn't needed, that's only required for non-essential cookies (login cookies are essential). For example if you added ads to the site, you'd need the notice since those almost always use tracking cookies.
- SVG (or even PNG) logo would be much appreciated, perhaps in various different layouts (e.g. logo only, logo + text, etc). I couldn't find the original art for the snark so I ended up clipping it out from the original site's graphics.
- Reordering the sections on mobile might also require reordering on the desktop layout as well, but I'll need to try it first to see.
- I think I might have not implemented the "star maps" feature, should I add it in?
- I can add the 'other users online' section back in, maybe it can just show logged in users based on their last activity? If they visited any page on the site in the last 10 mins maybe? Tracking the anonymous logins is a lot harder, and I'd like to avoid that if possible. Same for the link to the dedicated page of online users - it seems somewhat overkill. Maybe the whole thing could be replaced with an "active users" section on the home page, like TWHL has?
- The forum thread icons just represent if there's new posts that were posted after your last login. I can add a tooltip to them.
- BBCode - the old site had some weirdness going on with bbcode, especially in articles - some of it was pre-parsed and the html saved to the database, and some of it was parsed and converted into html when the page is loaded. It's more consistent now, the bbcode is saved to the database and also converted to html and that's saved into a second database field. It's a little bit more efficient than having to parse the bbcode for every post every time a page is loaded.
- The BBCode is actually using the same parser that I made for TWHL, which I call WikiCode. I removed most of the TWHL-specific features and added the missing SnarkPit features so it's pretty much compatible with how SnarkPit did it before. Markdown is probably not a good idea since there's a lot of history in the threads that use BBCode stuff that markdown doesn't support too well. My WikiCode variant is sort of a mashup of BBCode and Markdown which allows stuff like `*bold*` and `/italics/` without typing `[b]bold[/b]` and so on, so hopefully it does the trick.

### Checklist 

- [ ] Increase font size and review pages to make sure nothing looks weird
- [ ] Tweak mobile slideout menu
- [ ] Home - mobile updates
    - [ ] Center content
    - [ ] Reorder sections
    - [ ] Change spotlight to smaller icons in mobile
- [ ] Maps
    - [ ] Star maps functionality/filtering
    - [ ] Normalise map thumbnail heights like TWHL does (do this on user detail page also)
    - [ ] Sorting options for both recently created and recently modified
    - [ ] Add 'subscribe' button to map detail page
    - [ ] Update 'Download' button to 'SnarkPit Download'
    - [ ] Add link to game purchase / download page, update game label ("More content for:" instead of "game:")
- [ ] Admin stuff
    - [ ] Add game store link / download page to game config editing
- [ ] Tweak thumbnail size of article/download images in the index pages
- [ ] Add back in 'Online users' section (logged in users only? or active users widget on home page?)
- [ ] Forums
    - [ ] Add tooltips to forum thread icons
    - [ ] Tweak padding on user details in forum posts
    - [ ] Fix smilies in the live preview (its pointing to snarkpit.net instead of reboot.snarkpit.net)

**Comment by LogicAndTrick** [comment]
Progress in checklist above. Notes:

- For changing the order on the mobile layout, I was able to get most of the changes you wanted except for "Recent Updates" which I couldn't put after "Recent Maps" - I can only change the order of things within the same column, but I can't put stuff from the left column underneath stuff from the right column. I put it below "Spotlight" instead, hopefully that's enough. If you really want it that order it's possible, but I would need to duplicate that section into both columns and toggle which one is visible based on the window size - not ideal, but doable.
- I made some changes to the mobile home page - only the first news post is shown, and the first 3 spotlight items. Let me know if this is no good, I can change it back. I ended up changing TWHL to just one new post on the front page since we post news so rarely, and it doesn't make a lot of sense to make everyone scroll past multiple old news post to see the forum updates. Spotlight is 3 because 3 icons fits nicely into 1 row (4 seems a bit small, and 5 is way too small).

**Comment by Riven202** [comment]
Hey Dan,

Thanks for continuing to stay on top of this and keeping things moving! I noticed some of the updates, and I think its coming along nicely! I've added my responses below your comments and numbered them for ongoing reference. 

**1. SVG Graphics / Updated Logo:**
> SVG (or even PNG) logo would be much appreciated, perhaps in various different layouts (e.g. logo only, logo + text, etc). I couldn't find the original art for the snark so I ended up clipping it out from the original site's graphics.

   - I noticed the new SVG logo in the slide sheet, which is looking great! I might tweak it a little as I think the black border comes out a little more than I anticipated at a mobile scale, but will work on getting higher-res separated version of the background snark image to finish out the main header logo for 4K screens a little better. More on that later.

**2. Responsive Ordering:**
> For changing the order on the mobile layout, I was able to get most of the changes you wanted except for "Recent Updates" which I couldn't put after "Recent Maps" - I can only change the order of things within the same column, but I can't put stuff from the left column underneath stuff from the right column. I put it below "Spotlight" instead, hopefully that's enough. If you really want it that order it's possible, but I would need to duplicate that section into both columns and toggle which one is visible based on the window size - not ideal, but doable.
   - Since Journals were always an underutilized tool, I'd like to avoid giving them a spot at the very top of the home page. Can recent journals section go beneath recent maps section on the medium and large screen breakpoints otherwise? I like where it appears beneath the recent maps on the small screen breakpoint now. Does that still keep it in the same column? 

**3. Star Maps:**
> I think I might have not implemented the "star maps" feature, should I add it in?
   - Yes please. That's essentially our recommended maps tag. These are maps that are definitely worth a user's time. 

**4. Online Users:**
> I can add the 'other users online' section back in, maybe it can just show logged in users based on their last activity? If they visited any page on the site in the last 10 mins maybe? Tracking the anonymous logins is a lot harder, and I'd like to avoid that if possible. Same for the link to the dedicated page of online users - it seems somewhat overkill. Maybe the whole thing could be replaced with an "active users" section on the home page, like TWHL has?
   - If you're referring to the awesome active user list on TWHL like below, then yes! I think that would be an awesome addition and provide some much-needed encouragement for users ghosting. 
      - ![image](https://github.com/LogicAndTrick/snarkpit-reboot/assets/1342806/1a33669d-5185-413c-9cde-9187c273ac30)
      - Can we add that beneath the _Recent Forum Posts_ section on the home page? 
      - I'm completely fine with eliminating the dedicated user's list page otherwise. 
   - Additionally, can we add a new line to the common footer on each page?: 
      - ![image](https://github.com/LogicAndTrick/snarkpit-reboot/assets/1342806/23b97a06-d56a-4fb7-b9f9-0b5a1c10754d)
     - Into something that reads like:  
        - © Snarkpit.net 2001 - 2024 • about us • contact
           184 users online • 13 members browsing forums
           Snarkpit v7.0 created this page in 1.0037 seconds.

**5. Forum Icons:**
> The forum thread icons just represent if there's new posts that were posted after your last login. I can add a tooltip to them.
   - I think that's completely fine and does the trick!

**6. News Items:**
> I made some changes to the mobile home page - only the first news post is shown, and the first 3 spotlight items. Let me know if this is no good, I can change it back. I ended up changing TWHL to just one new post on the front page since we post news so rarely, and it doesn't make a lot of sense to make everyone scroll past multiple old news post to see the forum updates. Spotlight is 3 because 3 icons fits nicely into 1 row (4 seems a bit small, and 5 is way too small).
   - I think that's a good idea, especially for mobile. I know the "Community News" header at the top is a link to more news, but should there be a new button added beneath that post in the mobile version to "_See more_" ? But maybe that's not necessary. What are your thoughts?

**7. BB Code:**
> BBCode - the old site had some weirdness going on with bbcode, especially in articles - some of it was pre-parsed and the html saved to the database, and some of it was parsed and converted into html when the page is loaded. It's more consistent now, the bbcode is saved to the database and also converted to html and that's saved into a second database field. It's a little bit more efficient than having to parse the bbcode for every post every time a page is loaded.
The BBCode is actually using the same parser that I made for TWHL, which I call WikiCode. I removed most of the TWHL-specific features and added the missing SnarkPit features so it's pretty much compatible with how SnarkPit did it before. Markdown is probably not a good idea since there's a lot of history in the threads that use BBCode stuff that markdown doesn't support too well. My WikiCode variant is sort of a mashup of BBCode and Markdown which allows stuff like *bold* and /italics/ without typing [b]bold[/b] and so on, so hopefully it does the trick.
   - I think the new system works great, and don't have any other objections to what you pointed out. I think the new system makes a lot more sense for what we need.  

Will follow-up with more comments.

**Comment by Riven202** [comment]
Hope all is well, Dan! I've got another round of comments queued up, but I think this will cover the last of the design changes, and the rest should just be bug squashing. There was a bit of time between me initially testing the site to write these down, and me posting them here, so I noticed some were no longer issues, and appreciate you continuing to work on the site!  I'm feeling really good about how both the mobile and desktop versions of the site are coming along. And it's performing beautifully for me! It's so wonderful to see the Pit get the kind of love needed to keep the content around. I really appreciate your time on all this. 

Here's a design thoughts I had as I made efforts to check out both the mobile and desktop versions of the site:

## Design Requests:
1. **Header Menu Links:** 
    ![image](https://github.com/user-attachments/assets/fee9e529-897a-4112-9888-3c712a2932a2)
- The top menu links are noticeably different from the original website's link font, which I think will remain distracting to some past users. Does it break anything to update the font for those links to the standard web font: _Impact_ at around 26 pt size, per my experiments below? 
    ![image](https://github.com/user-attachments/assets/dfc7cfd5-d442-4e00-b3fa-9ba696610ca1)
2. **Side Sheet:**
- Can the menu items in the side sheet also match the font and style (all-caps) of the main menu? 
3. **Threads:**
- Looks like some users have written URLs and other long strings in their user profile fields that appear next to their avatar on thread posts. Does it make sense to add a truncation script with an ellipsis to each of those items so as to prevent them from overrunning into the body of a post? I'd expect the following example might look something like: _"https://www.gamebanana.c..."_
   ![image](https://github.com/user-attachments/assets/4a415eb1-e70a-474d-989f-f9c430118e84)
4. **Forum Icons:**
- On a forum's top page with the list of icons along the left side, can we have the icons stack vertically rather than horizontally? They take up a lot of room on mobile, and can make reading the forum topic rather difficult: 
   ![image](https://github.com/user-attachments/assets/9f337e55-381c-4d5b-bb54-b61edb24e1f4)
5. **Thread Posts on mobile:**
- On the mobile post shown below, the default author section for each post takes up an inordinate amount of space, which can make for a lot of scrolling through a longer thread: 
   ![image](https://github.com/user-attachments/assets/721483d8-acbd-451d-b863-e9debaec36d0)
   - Does it make sense to provide a different stacking arrangement on mobile breakpoints that reduce or limit some of the author's details? I really appreciate how it was accomplished on TWHL (highlighted below), and I can see that working on the Pit in a similar fashion: 
   ![image](https://github.com/user-attachments/assets/de65e3a6-b3f8-46e7-99a2-dc7e32554866)
   - The SnarkPit always listed a lot with a user's post, notable even back on the early 2000's internet standards, so on mobile I would appreciate limiting most of those details in favor of a slimmed down thread to reduce scrolling. I still like the idea of listing their post quantity beneath their name, since that conveys some kind of seniority across user's feedback. Formatted, I can see the following info appear at each author's section above the post itself: 
   - Riven
   - 1,628 posts
   - Posted on Sun Jul 7th 2024 at 4:22pm 
- I imagine the author's username would be a little larger font size than the other details. 
6. **Thread Edit Buttons on Mobile:**
- The edit buttons on mobile when logged-in wrap in a way that looks odd or may take up more room than necessary in the overall height of each post. I understand this may only be an issue for the super admin profile, since that account is the only one that sees a delete button on every single post, so consider this a low-priority. 
7. **Map Details Page:**
- Since there was always a bit of inefficiently distributed spacing for a Map's detail page, does it make sense to reorder or sort some of the sections to give more room to the Map Info section, as shown below?
   ![image](https://github.com/user-attachments/assets/35e8c450-53ce-44a3-9750-51123bd15e3d)
   - I imagine this would allow the Map Info section to grow into 2/3rds of the space provided ideally. 
   - I'm ok if this means the map rating section needs to come before the downloads section in all breakpoints as they appear on the detail page. 
- Can we add a breadcrumb trail feature at the top of a map's detail page that gives you a link back to the Map's search page with that game category filtered? This would work in a similar fashion to the forum breadcrumb trail feature. So on a map it might read like: _"Maps / Counter Strike Global Offensive / surf_a_xbl..."_ 
   - I can image the map name itself might need a truncation with ellipses added to prevent the breadcrumb trail from getting too long: 
   ![image](https://github.com/user-attachments/assets/8964fc27-6ace-42f7-844e-73d13805ee85)
  - I understand we already have links back to the map's game page, but this would standardize the breadcrumb feature in a way that makes it obvious no matter where you are on the site. 
8. **About Us:**
- Have you visited the About us page (linked in the footer on each page)? Don't forget to add your own entry there, since you've more than earned it! Each developer has had an opportunity to leave their mark there no matter how promising or real you feel it might need to be. I like the sense of history it provides, and a bit of insight into the story of the Pit itself. 
9. **Thread Reply Section on Mobile:**
- While I love how true-to-form you've been able to design and modernize the reply section on SnarkPit, I'm realizing on mobile breakpoints the Smilie section always left much to be desired. 
   - The Smilies sections takes up more than its fair share of space on mobile. Does it make sense to collapse the list to a button on mobile similar to the other icon buttons throughout the rest of the site? 
![image](https://github.com/user-attachments/assets/3ca50fef-6d42-4c52-91bc-383557b77186)
   - I imagine using this icon or one similar in a button would be sufficient: https://fontawesome.com/icons/face-smile?f=classic&s=regular
   - Additionally, thinking through the location of the "Add signature" checkbox when the message preview can get quite long on mobile, does it make sense to move that checkbox together with the "Live Preview" checkbox so that its not as easy to miss?
   - If we also collapse the Smilies section, then I can image both the message preview and editing field would grow to fill the width like so: 
      ![image](https://github.com/user-attachments/assets/1ec659d5-05eb-4548-a5a1-e56f313242cb)
   - This way, when the Smilies button is expanded, it would fill the space in the way the reply section is currently formatted. What are your thoughts on this? 



## Bugs:
1. **Header:**
- At medium breakpoint, the top menu expands to wrap links (links that appear when logged-in), causing the SnarkPit logo graphic to split: 
  ![image](https://github.com/user-attachments/assets/5554576b-3fde-4133-bd4b-ab0d4508e846)
2. **Thread Quote Button:**
- At some point when clicking the quote button on a thread to reply, the page seemed to time-out and never load up. This however, seems to have been fixed between the time I noted this, and posting it today. 
3. **Thread Edit Buttons on Mobile:**
- The edit buttons on mobile when logged-in wrap in a way that looks odd or may take up more room than necessary in the overall height of each post. I understand this may only be an issue for the super admin profile, since that account is the only one that sees a delete button on every single post. So consider this a low-priority. 
4. **Maps Pagination on Mobile:**
- Maps pagination runs off the edge of the screen when viewed at mobile breakpoint: 
   ![image](https://github.com/user-attachments/assets/2a9447b5-f197-4d04-a96f-7782ece45703)
   - I'm not sure if this means the pagination list is simply shorter at smaller breakpoints, revealing fewer pages to cycle through. What are your thoughts on that? 
5. **Map Detail Page Image Buttons:**
- On first load of a Map's detail page, the image cycle buttons beneath the image appear correctly sized. However, on tapping or clicking one of them, they become a much larger size as shown below: 
   ![image](https://github.com/user-attachments/assets/8ca05283-68ac-4683-8c93-22ac5b200987)
   ![image](https://github.com/user-attachments/assets/7694e609-b1af-4a6e-8222-560c158c2659)
- I don't necessarily mind the larger button sizes on mobile, as those appear correctly sized for tapping with a finger. I suspect however, the resizing behavior should not be dependent on clicking/tapping one of the buttons, but rather based on real-time breakpoint resizing?

## Items I'm working on: 
- [ ] New .svg icons for each game
- [ ] higher resolution version of the snark logo on the main header. 

I will continue to review more of the site testing each of its features and will follow-up here. I'll start to limit my comments to bugs and no longer anticipate asking for any new feature requests like what was listed above, so we can start to wrap-up what we have. 

Looking forward to hearing back!

**Comment by LogicAndTrick** [comment]
Thanks for the detailed feedback, those are all achievable - I'll probably work on them this upcoming weekend.

**Comment by LogicAndTrick** [comment]
Ok, I've worked through most of the most recent feedback - I'm aware I've still got some more things to do from the initial post.

Design requests:

- [x] Header menu links
- [x] Side sheet links
- [x] Truncate long text in user details with ellipsis
- [x] Thread list icons (will be vertical at mobile breakpoints, horizontal on larger screens)
- [x] Post user details compressed on mobile (I played with this a lot to get something that looks good, see what you think)
- [x] Post edit/delete buttons wrapping on mobile (I tried to fix this when updating the post user details for mobile)
- [x] Breadcrumbs on map details page
- [ ] Add my info to about page (I'll do this once it goes live :)
- [x] Collapse smilies on mobile (I did this using a dropdown similar to how TWHL does it, but also on mobile I removed the 'show more' option because some of those smilies are rather large and cause the expanded section to be way too big on mobile)

Bugs:

- [x] Top menu wrapping at medium desktop resolution
- [ ] Thread quote button causing timeout (not sure about this one, quote button shouldn't change the page, but it does do a query to the server to get the bbcode text to quote)
- [x] Post edit/delete buttons wrapping on mobile (already mentioned above)
- [x] Reduce number of pages in pagination for mobile (this was difficult, there's only so much I can do without redoing a lot of stuff - it'll show the next/back buttons and 2 pages either side for mobile now)
- [x] Image back/next buttons resizing on mobile (I just made them the larger size on every breakpoint)

**Comment by LogicAndTrick** [comment]
@Riven202 long time no update :)

Someone nudged me about this today as the main site was offline (apache had crashed, I fixed it already) so I figured I'd close out the two last things that I hadn't done yet. No pressure, but it'd be great to get the new site deployed at some point. Let me know if you have more feedback or if there's any feedback I missed in the thread.

Added today - I think this was the last remaining things that weren't done yet:
Active users on front page
![image](https://github.com/user-attachments/assets/1eae878a-6cdf-48fe-bb49-22f397c19d3f)

Online users in footer - tracking anonymous users is something I would like to avoid, so the footer will only show the count of registered users who have accessed the site in the last 10 minutes - it is cached and only updates once a minute so don't worry if you log in and it doesn't immediately go up by 1.
![image](https://github.com/user-attachments/assets/5e3bebd9-63e0-4e4f-a020-c862b5426ce3)

**Comment by Riven202** [comment]
Hey Dan,

Thanks for soldiering on through the long months! It feels as though we're nearly there. And with the recent Half-Life 2 20th anniversary making a splash in the Valve community, it'd be nice to get the new Pit up and running in time for the holidays this December!

The site is looking slick, and running even better, and the new Active users feature is awesome! It really sells that community energy when you can see others engaging in real time. Thank you for implementing it!

I have just a few more bugs, and some hanging design "cleanup" requests, since I made it a point to navigate the site entirely on mobile, as well as push and turn all the back-end dials you painstakingly re-crafted and re-organized. The site is humming super smoothly, so this is my last list of design items, and with your thumbs-up, I'm comfortable going public with v7. Let me know if you have any objections.

### A few notes on my previous promises: 
 * I will probably not get around to producing new .svg icons for the games. And since I noticed you provided an easy process to implement that, no need to wait on this. I'll likely piecemeal those updates over time. 
 * The header Snark graphic itself if sufficient for now, and probably won't get replaced any time soon. However, I'd like to see the new vector text logo and subtitle replace the raster version on the header. Though, I'm not sure how distracting it will appear since it's a bit brighter it seems. 


## Design Notes:
1. **Menu button overlaps header logo graphic on mobile:** 
    Note: This is not evident for whatever reason when I  emulate a mobile responsive simulation with the dev mode in Edge. So surprising to see it this way:

    ![image](https://github.com/user-attachments/assets/a98df1ed-5729-4199-98f2-35d7f817370f)

2. **Poster Listing Squeezed on Mobile:**
    When viewed on my iPhone Xs, the overlap of the post count above the post timestamp gets a little squashed.  I'm ok hiding the post count on mobile if it means we don't get an awkward blank line between the user's name and the timestamp, since there's many users with usernames just as long as this one. 

    ![image](https://github.com/user-attachments/assets/ebc7c5a2-bf70-4eff-9a89-5377c1ba5e78)

3. **Home Page Journals section:** 
    a. I may have missed where you previously responded to this, but since the Journals feature was always underutilized, I'd like to avoid giving them a spot at the very top of the home page, especially since they may perpetually remain dated.
    b. Can we slip this to the bottom of the maps section at medium and large breakpoints? Otherwise, I like where it appears beneath the recent maps on the small screen breakpoint now. 
4. **Access Levels:** 
    Do Moderator, Admin, and Super Admin provide distinctly different access levels? Admin and Super Admin seem to have similarly listed site management options, but I may be missing something. 
5. **Snow Snarks:**
    a. This is an old fan favorite from the very first version of the Pit. They are small images of snarks with santa hats on that gently drift down from the top of the page to the bottom. Clicking on one causes it to play an explosion gif. I enable them manually through the Super Admin > Site Config page under: "snow_snarks" parameter. I usually do this at the beginning of December each year. 
    b. Since this almost always gets cheers, is it easy to re-implement this feature again? 
6. **Upgraded Search:** 
    I'm loving the new search results! It's actually useful, finally!
7. **Support Forum Question / Answer Feature:**
    a. The v6 edition of the Pit introduce a "Support Forum" which meant the ability for the original poster, or a moderator to select a right / wrong answer on any subsequent post in that thread. That post receives an accepted answer banner. And the option was a drop-down list on each post. The options are: 
      * Unset status
      * Assisted
      * Incorrect
     
    b. These threads could be filtered in a search. 
    c. That forum's thread list also included a column showcasing threads with answers:

    ![image](https://github.com/user-attachments/assets/8095627d-6893-400a-b4cf-260f29f0d51d)

    d. Image below is what an accepted post looks like, with the drop-down list at the top. 
      * I am not sold on this graphic design for an accepted answer, because I believe it's easy to miss while scrolling down a long thread, but it worked. 
      * Would be nice to have a "jump to first accepted answer" button or something similar at the top of such a thread. 

    ![image](https://github.com/user-attachments/assets/7deae08a-cbfe-40c3-9a9d-db66002fa375)




## Bugs: 
1. **Header Banner No Link on Mobile:**
    The Navbar banner logo image does not work as a link back to the home page at mobile screen breakpoints. 
2. **Missing Reply Text Formatting Buttons on Mobile:**
    The reply section hides any text formatting menu buttons while viewed on mobile responsive size: 
    ![image](https://github.com/user-attachments/assets/64cddec3-7ac9-4d56-9846-ff0a73469ab8)

3. **Formatting help Button:**
    Clicking the "_Formatting help_" button in the reply section directs you to a 404 section page now. I'm not actually sure what that page used to be though. Do you have a new guide that gives direction on how to use the new text styling syntax and commands? 
4. **Forum Thread List Spillover:**
    a. The Forum thread list on any forum page on mobile overflows outside the mobile width for me on iPhone Xs. 
    b. Example taken from the _Forums / General Banter_ thread page: 
    c. This same issue also appears on the _Forums / Threads_ page when accessed from a user's control panel _Links_ list. 
    ![image](https://github.com/user-attachments/assets/9635d6e4-0c31-43a9-b77c-bdda53345182)

5. **Post List Mobile View Not yet Updated:**
    a. _Control Panel > View Forum posts_ option takes you to a post list that is not yet updated to match the recently revised mobile version of the post list mobile responsive layout. 
    b. _Forum / Posts:_
    ![image](https://github.com/user-attachments/assets/f7c96a90-884f-4e49-8f23-2e167f397892)

6. **Change Email SQL Error:** 
    a. _Control Panel > Update email_ option takes me to an SQL error page:
    ![image](https://github.com/user-attachments/assets/5f28959d-b63d-4096-89bc-73ece2215316)

7. **Spotlight List Spillover:** 
    a. _Admin page > Spotlight > Spotlights table_ overflows beyond the edge of the frame on mobile:
    ![image](https://github.com/user-attachments/assets/d5f3d491-943b-4094-ad82-d6afb59e6e24)

8. **Games List Spillover:**
    a. _Admin page > Games > Games table_ overflows at most all breakpoints. 
    b. Since the table is so wide, maybe it makes sense to give it an in-frame horizontal scroll bar?

### Conclusion: 
...And that's it! The site is running extremely smooth for me; definitely makes it feel much more energized! Looking forward to being able to reliably use the site on mobile now! 

Feel free to reach out to me on Discord if you have any questions/follow-up if we need to chat before taking next steps. Thanks again for putting all your time on this! It really means a lot that you're able to bring the old site back to life! Looking forward to getting feedback from everyone once we go live. 

Best,
Eric

**Comment by LogicAndTrick** [comment]
Design:
1. I changed it to hide the snark logo when the viewport is too narrow, I couldn't really think of any decent alternative to handle smaller screens that didn't look worse than that.
    ![chrome_s9Lbwda9uO](https://github.com/user-attachments/assets/89c85c1b-3031-43ed-acb2-d0b75d090782)
2. Slightly updated the layout on narrow screens to improve the view:
    ![image](https://github.com/user-attachments/assets/070be92b-6290-4d3f-80c2-e58a7ed1c979)
3. Moved journals to be below latest maps on all size screens
4. Access levels:
    - moderators can: delete/edit articles, forum posts, forum threads, downloads, and news
    - admins can: everything moderators can do, and use all parts of the admin page except for the super admin stuff
    - super admins can: every admins can do, change the level of other users, obliterate users (for spammers), delete user accounts (for non-spam account remove requests)
5. TWHL has something similar and its based on the date (displays during December) rather than having to be turned on, so I've done something similar for the snow snarks too. Add `?snow` to a url to preview outside of December. Unfortunately the explosion gif looks pretty bad in Firefox due to how it handles looping, I can't do anything about that...
6. Glad you like it :)
7. Hoping that we can exclude this feature as it wasn't used too much (only about 1 in 10 posts were marked as answered, and those forums were not used all too much)

Bugs:
1. Made the header a link for mobile screens
2. Unfortunately that's hard to change, there's just not enough screen page to put all the buttons there without it looking very messy, which is why those buttons are just hidden on small screens.
3. I haven't made this page yet but it'll be very similar to the [TWHL formatting help page](https://twhl.info/wiki/page/TWHL%3A_WikiCode_Syntax), with adjustments to match the Snarkpit flavour of the formatting.
4. I hid the 'views' and 'last post' columns on the thread list page for narrow screens, which makes it a bit nicer. On the page accessed from user control panel > links, the 'user' column is also hidden.
5. Updated the user > forum posts page to match the regular post listing page on mobile.
6. I can't seem to get this error when I try, can you try again?
7. To be honest I didn't design any of the admin pages for mobile since that's behind the scenes stuff, but I've fixed this by removing the 'by' column on small screens.
8. As above, but fixed by tweaking the text wrapping and hiding some columns as the screen shrinks.

**Comment by Riven202** [comment]
Hey Dan,

It's been almost a year since we spoke. My apologies for letting it drag on as long as I have. I realize you may have already moved on and we may not have an opportunity to complete this project with you. So please let me know if we need to bring someone else on. I did not intend to step away right at the finish line. 

That said, I think the reboot site is the best version of the Pit yet, and your work to resurrect it again is sooo greatly appreciated! It's looking amazing, and I think the work you've done is incredible! 

I've noted the follow-ups to your last few comments above. **_But I believe the Reboot is ready to go live when you are!_** Otherwise, we can discuss separately what the next steps ought to be. 



### Previous Design Comments Follow-up:
**Item 5: Snow Snarks:** LOVE that you were able to implement these! Can't wait to see them on the new site! They work even better than before it seems. I like the varying sizes of them too. This is great! 
**Item 7: Support Forum Question / Answer Feature:** Understood this was not as widely used as I suspected, and agree it's likely not worth the effort to re-implement. 
#
### Bug Update feedback:
**Item 6: User Update Email SQL Error:** Link is working for me now. Cannot reproduce the issue myself. Could have been a one-off. 

Looking forward to hearing back!

**Comment by LogicAndTrick** [comment]
Nice! I'll plan to do one final refresh of the reboot database (probably next weekend), and if everything looks good, we can switch the URL across.
